docs/DocGenerator/Config.py

```diff
@@ -0,0 +1,127 @@
+            if 'recursive' in include:
+                element.recursive = include['recursive']
+            self.include_paths.append(element)
+
+    def __read_include_regex(self):
+        if not 'Include regex' in self.__config_data:
+            raise Exception("Include regex is empty")
+        self.include_regex = self.__config_data['Include regex']
+
+    def __read_ignore_paths(self):
+        if 'Ignore paths' in self.__config_data:
+            self.ignore_paths = self.__config_data['Ignore paths']
+
+    def __read_valid_tags(self):
+        if 'Valid tags' in self.__config_data:
+            self.valid_tags = self.__config_data['Valid tags']
+
+    def __read_required_fields(self):
+        if 'Required fields' in self.__config_data:
+            required_fields = self.__config_data['Required fields']
+            if 'Module' in required_fields:
+                for required_module_field in required_fields['Module']:
+                    self.required_fields.module.append(required_module_field)
+            if 'Tests' in required_fields:
+                for required_tests_field in required_fields['Tests']:
+                    self.required_fields.tests.append(required_tests_field)
+            if 'Case Sensitive' in required_fields:
+                self.required_fields.case_sensitive = required_fields['Case Sensitive']
+
+    def __read_ignored_fields(self):
+        if 'Ignored fields' in self.__config_data:
+            ignored_fields = self.__config_data['Ignored fields']
+            if 'Module' in ignored_fields:
+                for required_module_field in ignored_fields['Module']:
+                    self.ignored_fields.module.append(required_module_field)
+            if 'Tests' in ignored_fields:
+                for required_tests_field in ignored_fields['Tests']:
+                    self.ignored_fields.tests.append(required_tests_field)
+            if 'Case Sensitive' in ignored_fields:
+                self.ignored_fields.case_sensitive = ignored_fields['Case Sensitive']
+
+    class __paths:
+        def __init__(self):
+            self.path = []
+            self.recursive = True
+
+    class __fields:
+        def __init__(self):
+            self.module = []
+            self.tests = []
+            self.case_sensitive = False
+
+
+conf = Config()
+
+print(conf.project_path)
+
+print(conf.documentation_path)
+
+for element in conf.include_paths:
+    print(element.path)
+    print(element.recursive)
+
+for element in conf.include_regex:
+    print(element)
+
+for element in conf.ignore_paths:
+    print(element)
+
+for element in conf.valid_tags:
+    print(element)
+
+print(conf.required_fields.case_sensitive)
+for element in conf.required_fields.module:
+    print(element)
+for element in conf.required_fields.tests:
+    print(element)
+
+print(conf.ignored_fields.case_sensitive)
+for element in conf.ignored_fields.module:
+    print(element)
+for element in conf.ignored_fields.tests:
+    print(element)
```

Comment:
I would remove from the `conf` object creation to the end.

Reply:
This part of the code is implemented for development purposes and will be deleted when it is fully integrated, Being able to be tested in the full solution.